### PR TITLE
Documentation improvements and minor tweaks to the script

### PR DESCRIPTION
**Pull request by ekuns**
Three commits to my fork are included here:

Enhance the documentation
More doc improvements and typos fixed in script
Parameterize the region from which to get the list of AZs in each region

### CODE REVIEW

**Comment by ekuns** [comment]
Hmm, I thought to make another pull request for my last change to allow credentials for some accounts to be bad or missing and have the script continue processing.  But it looks like that last 4th commit automatically got included into this pull request.  Let me know if you don't see 4 commits as part of this pull request.

File: example.output [hunk 1]
```diff
@@ -1,9 +1,9 @@
 Mapping for region ap-northeast-1
-+-----------------+---+---+---+
-| xxxxxxxxxxxxx   | a | c | d |
-| yyyyyyyyyyyyyyy | a | c | d |
-| zzzzzzzzzzzzz   | a | c | d |
-+-----------------+---+---+---+
++-----------------+---+---+---+---+
+| xxxxxxxxxxxxx   | a | . | c | d |
```
Comment (MarkBiesheuvel) [inline example.output:3]: @ekuns Were you able to find such an example with the script? :)

File: script.py [hunk 2]
```diff
@@ -66,22 +69,30 @@ def get_regions(self):
...
-            OfferingType=OFFERING_TYPE,
-            ProductDescription=PRODUCT_DESCRIPTION,
-        )
-
-        return {
-            offering['ReservedInstancesOfferingId']: offering['AvailabilityZone'][-1:]
-            for offering in response['ReservedInstancesOfferings']
-            if 'AvailabilityZone' in offering
-        }
+        if self.name in bad_credential_list:
```
Comment (MarkBiesheuvel) [inline script.py]: @ekuns Could you replace the `bad_credential_list` with a property like `self.valid_credentials` which is set to `True` in `__init__` and sets it to `False` when it catches a `NoCredentialsError`?

File: example.output (hunk 1 above)
Comment (ekuns) [reply example.output:3]: Yes, except for changing the names, that output is taken directly from output when I run it against a few of the accounts I have access to.

**Comment by ekuns** [comment]
Here is the output from a group of accounts I tested with only the names changed:
```
+----------------+---+---+---+---+
| aaa   | a | . | c | d |
| bbb   | a | . | c | d |
| ccc   | a | . | c | d |
| dddd  | a | b | c | d |
| eee   | . | . | . | . |
| fff   | a | . | c | d |
+----------------+---+---+---+---+
```
The account `eee` above doesn't have valid credentials for it because it's only used in reference to a role change somewhere.  Something like that anyway.

File: script.py (hunk 2 above)
Comment (ekuns) [reply script.py]: Done!  I'm still learning Python so my code isn't always very Pythonic.  Your asking me to make the change helps me get better at writing Python the correct way, so thanks for that!

**Comment by ekuns** [comment]
OK, how's this group of five check-ins on my side?

**Comment by MarkBiesheuvel** [comment]
Glad to hear that you found a great example amongst your accounts.

And thanks again for the effort, looks good 👍

**Pull request merged**